pyproject.toml

```diff
@@ -29,13 +22,13 @@ classifiers=[
 ]
 dependencies = []
 description = "A pure Python package for reading and writing DICOM data"
-#  dynamic = ["version"]  uncomment for pydicom v3.0
+# dynamic = ["version"]
 keywords = ["dicom, python, medical, imaging"]
 license = {text = "MIT"}
 name = "pydicom"
 readme = "README.md"
-requires-python = ">=3.7"
-version = "2.4.0" # remove for v3.0
+requires-python = ">=3.10"
+version = "3.0.0.dev0"
```

Comment:
Not sure what you mean here.  I've changed things so that we only have one place where version is set - and it is in `pyproject.toml`.  Then in theory `pydicom.__version__` is not needed as anyone can get all the metadata using `importlib`.  But I set it in `_version.py` according to the metadata, for backwards compatibility.

If you mean in terms of using `dynamic` rather than a literal version, I so far haven't been able to get that to work, flit tries to import pydicom and wasn't able to do some of the sub-imports.  There are also build backends that can pull the version from source control, so maybe that could be put in place at some point.